### ISSUE TITLE
[SPIKE] Investigate Tugboat upgrade to Debian Bookworm

### ISSUE THREAD

**Issue by ndouglas** [department-of-veterans-affairs/va.gov-cms#14075]
## Description

Debian released Debian 12 "Bookworm" [on June 10](https://www.debian.org/releases/bookworm/). We should attempt an experimental upgrade to Bookworm on Tugboat to identify improvements and issues we might face in a wider upgrade.

## Tasks
- [x] Switch "Bullseye" to "Bookworm" in the Tugboat config.
- [x] Push a branch with this change upstream to the va.gov-cms repo.
- [x] Create a base Tugboat preview based on that branch and set it so that child branches use this base preview.
- [x] Create a va.gov-cms PR with some trivial change (touched text file, etc) on a child branch.
- [ ] Verify that tests pass and the site works normally.
- [x] Document findings in this thread.
- [ ] Open followup ticket(s) to address issues and/or perform upgrade.

## Acceptance Criteria
- [ ] Above investigatory process has been carried out.
- [ ] Findings documented in this thread.
- [ ] Followup ticket(s) opened to address issues/perform upgrade.

**Comment by ndouglas** [comment]
This is blocked by a couple things:
- I think we need SOCKS access, which I think is blocked by Platform onboarding and PIV card.
- Tugboat doesn't yet have a Bookworm PHP Docker image available.

I've checked into the latter, but we might need to find a replacement for this ticket. Nevertheless, I did do a limited show-and-tell of Tugboat in a call, so progress may have been made 🙂

**Comment by laflannery** [comment]
Blocked by lack of PIV access

**Comment by ndouglas** [comment]
This is unblocked -- Tugboat has PHP images for Bookworm now.

**Comment by teeshe** [comment]
python won't install from the repository "apt-get install python" working with tugboat/bookworm container image. python2 package is listed for preview init in tugboat config but it seems the support for python2 was dropped for bookworm. I got below error when I tried building a preview and also when I exec into the container locally.

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBbmRFIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--bcb5f3dd64fc0301b990be64e5218c8bd62b3796/image.png)

**Comment by ndouglas** [comment]
@teeshe Are we stuck on Python 2 for some reason? I think I recently removed the only Python code in our codebase, but there might be a tool or something else we're using somewhere that's dependent upon it 😬

**Comment by ndouglas** [comment]
What happens if you sub `python3`?

**Comment by teeshe** [comment]
That should install/update fine. python3 ships with bookworm containers images. 
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBcFpFIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--8811568fbdb94dac74435422d6a168e82b0c55b1/image.png)
I'm not sure if we have some python2 scripts. That requires the package as a hard dependency.

**Comment by ndouglas** [comment]
I ripped the last Python code (which was just a test) out of va.gov-cms a while back, and I don't think there's any in `content-build`. Might be an indirect dependency through some other package, but if so, we should figure it out sooner rather than later so we can deal with it.

Might be worth:
- subbing `python3`
- removing any Python install 🤷🏻

**Comment by teeshe** [comment]
I wasn't able to get memcached to install on bookworm image; 'yes '' | pecl install -f memcached'
Error
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBdjFiIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--dd2f489f24f45d2709292a53aaf676c067a6b2a0/image.png)

multiple users have complained about same issue not working with Debian12. I came across this not sure how efficient this is yet.
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBZ3hjIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--43a6015c24caf65d3846180a19fecb9dd2d246ba/image.png)

Docs referred that Bookworm/Debian 12 should have more packages in addition to those supported from Debian11. same packages that installed into bullseye container are failing to work with bookworm.

**Comment by teeshe** [comment]
This would be completed this sprint @BerniXiongA6

**Comment by teeshe** [comment]
Finally got tugboat to work. Some breaking changes were initially seen after updating tugboat from bullseye to bookworm container image.
Observations: 
No support for python2 on bookworm. 
The version libmemcached-dev available on bookworm failed to install memcached using "pecl install -f memcached"
I had to replace "apt-get install libmemcached-dev" with "apt install -y libmemcached-dev zlib1g-dev libssl-dev"

changes are on branch [https://github.com/department-of-veterans-affairs/va.gov-cms/blob/VACMS-14075/.tugboat/config.yml](https://github.com/department-of-veterans-affairs/va.gov-cms/blob/VACMS-14075/.tugboat/config.yml)
I created base preview and created a child branch and built a preview from. [https://github.com/department-of-veterans-affairs/va.gov-cms/blob/feature/child-bookworm-test/.tugboat/config.yml](https://github.com/department-of-veterans-affairs/va.gov-cms/blob/feature/child-bookworm-test/.tugboat/config.yml)

Child preview was created from PR [https://tugboat.vfs.va.gov/6500a9012d3036648d18b9ad](https://tugboat.vfs.va.gov/6500a9012d3036648d18b9ad). something I did noticed was that preview was with a very light size about 150MB. But the preview seems to work fine so far.
